### ISSUE TITLE
url push with 1-click retrieval step consume two views 

### ISSUE THREAD

**Issue by AkaGabriel32** [pglombardo/PasswordPusher#2225]
## 🔬 How To Reproduce

1. create a url push
2. Check "Use a 1-click retrieval step"
3. Configure a single view
4. Click "push it"
5. Try to view for the first time the URL 

### Environment

- pwpush.com
- Android firefox

## 📈 Expected behavior

Secret url displayed

## 📎 Additional context

A message indicates that the secret has expired. 

In pwpush.com the audit screen shows one view ok and a second view expired at the same time.

**Comment by pglombardo** [comment]
Hi @AkaGabriel32 - I just tested this on my android phone with firefox and couldn't reproduce it.  Does it happen for you with another browser or device?

**Comment by AkaGabriel32** [comment]
Hi @pglombardo ;)
I have just tested with a second phone (android, xiaomi redmi note 10, firefox): problem reproduced.
I have just tested with an Ubuntu LTS 24.04 PC, Firefox: problem reproduced.
![audit](https://github.com/pglombardo/PasswordPusher/assets/172073744/6a94ecf2-ecd1-4196-b5fb-af8c3198255a)
![config](https://github.com/pglombardo/PasswordPusher/assets/172073744/aad418a6-7f9e-45d6-bdbc-78ee4e243341)
![error](https://github.com/pglombardo/PasswordPusher/assets/172073744/15d5f371-e581-40aa-9fff-04ecd8fd0275)
![firefox](https://github.com/pglombardo/PasswordPusher/assets/172073744/f858c432-b737-418c-a4f9-7f12f00c3fd5)

The problem only occurs if we choose "Use a 1-click retrieval step". In this case each view is counted twice (for my tests).

**Comment by pglombardo** [comment]
Hi @AkaGabriel32 - thanks for the helpful info.  I don't doubt this is happening for you.  I do think that something is double fetching.  Maybe like a link preloader... not sure.  Or maybe something else...

I'll try again to reproduce this soon.

**Comment by pglombardo** [comment]
Just got a second report of this:

> Looking at the logs, it looks like it immediately burns 2 views each time. My hunch is that it has something to do with macOS? I’m on macOS Sonoma 14.5. I’ve tried this with Safari, Brave and Orion, and in private and non-private windows. I’ve also tried turning off the macOS “Private Relay” feature. Yet, it always burns 2 views for me, and the log always shows a 1-2 second delay between the two (but I never get to see the actual link as the user)

Since this user validated on so many browsers, it has to be something global to MacOS - potentially antivirus?

The original report of this issue happens on Ubuntu & Android.

If we can identify who/what is prefetching links I can probably figure out a way to prevent it.

**Comment by wedowhateverwewant** [comment]
can confirm this is still happening on the latest version. URL with a single view with 1-click enabled is expired before even accessing the link. using the same windows 11  system with chrome and edge in private mode to avoid extension's it still expires. but passwords and files are just fine.

**Comment by pglombardo** [comment]
I _think_ I might have found a solution to this in #2650.  I'll release soon.

**Comment by pglombardo** [comment]
[v1.47.1](https://github.com/pglombardo/PasswordPusher/releases/tag/v1.47.1) is building now.  When possible, could some of you check if this issue is fixed?  Much appreciated.

**Comment by wedowhateverwewant** [comment]
Im able to still reproduce the issue on v1.47.1

**Comment by pglombardo** [comment]
Could you post the stack trace?

**Comment by wedowhateverwewant** [comment]
you mean the container logs?

**Comment by pglombardo** [comment]
My bad @wedowhateverwewant - I was on mobile and thought this was another issue.

**Comment by pglombardo** [comment]
I put out another attempt to block browser pre-fetching in [v1.49.3](https://github.com/pglombardo/PasswordPusher/releases/tag/v1.49.3) which is building now.  Could any one of you retry with that version and let me know?

Alternatively, you can try that version on [https://oss.pwpush.com](https://oss.pwpush.com).

Thanks!

**Comment by pglombardo** [comment]
Hi all - I finally found the root cause of this issue randomly while late night hacking.  The fix has been released in [v1.50.6](https://github.com/pglombardo/PasswordPusher/releases/tag/v1.50.6).

tldr; there was some JS code that was preloading the link for performance and eating up a view... which is now disabled.

Thanks all for the patience - this was a tricky one.

I'll close this issue out but if anything remains, please let me know and I can re-open if needed.
